hotdoc/utils/utils.py

```diff
@@ -148,22 +149,23 @@ def __get_extra_extension_classes(paths):
     Banana banana
     """
     extra_classes = []
-    wset = pkg_resources.WorkingSet([])
-    distributions, _ = wset.find_plugins(pkg_resources.Environment(paths))
 
-    for dist in distributions:
-        sys.path.append(dist.location)
-        wset.add(dist)
+    sys_paths = set(sys.path)
+    extra_paths = set([Path(x) for x in paths])
+    sys.path.extend(list(extra_paths - sys_paths))
+    del sys_paths
+    del extra_paths
```

Comment:
Adding `pathlib.Path` objects to `sys.path` [is explicitly supported](https://docs.python.org/3/library/importlib.metadata.html#distribution-discovery) by `importlib.metadata`'s distribution package discovery, "even though such values will be ignored for imports." Since `importlib.metadata` doesn't have the same `WorkingSet` support as `pkg_resources`, that becomes our replacement for the custom `WorkingSet`.

I'm admittedly only 90% sure this actually works like it's supposed to. A real-world test from someone who uses `hotdoc --extra-extension-path` and/or `hotdoc --extra-extension` would not be unwelcome.

Reply:
> I'm admittedly only 90% sure this actually works like it's supposed to. A real-world test from someone who uses `hotdoc --extra-extension-path` and/or `hotdoc --extra-extension` would not be unwelcome.

I say that, but running `hotdoc.extensions.gi.utils.get_language_classes()` from the `master` branch in Python 3.11, and from this PR branch in Python 3.12, produces the same results. So, that's a good sign that the basic idea is sound.

It's mostly just the `sys.path` fiddling in `hotdoc.utils.utils.__get_extra_extension_classes()` that I'd like to see validated through real-world testing.

Reply:
This does seem to break `--extra-extension-path`, I put up a reproducer with instructions @ https://github.com/MathieuDuponchelle/my-extension , hopefully this makes it easy to fix :)

Reply:
@MathieuDuponchelle Thanks! I'll take a look. I've also since noticed that Meson's docs use some custom extensions, so their code might be useful in testing as well.

Reply:
@MathieuDuponchelle 
That repo was a huge help, thanks! I believe I've now tackled this.